mpu6050.c

```diff
@@ -0,0 +1,316 @@
+#include <linux/init.h>
+#include <linux/module.h>
+#include <linux/device.h>
+#include <linux/err.h>
+#include <linux/i2c.h>
+#include <linux/i2c-dev.h>
+#include "mpu6050-regs.h"
+
+
+struct mpu6050_data {
+	struct i2c_client *drv_client;
+	int accel_values[3];
+	int gyro_values[3];
+	short temperature[2];
+};
+
+static struct mpu6050_data g_mpu6050_data;
+
+static int mpu6050_read_data(void)
+{
+	short temp, TEMP_OUT_H, TEMP_OUT_L, count;
+	int integer = 0;
+	short rest = 0;
+	struct i2c_client *drv_client = g_mpu6050_data.drv_client;
+
+	if (drv_client == 0)
+		return -ENODEV;
+
+	/* accel */
+	g_mpu6050_data.accel_values[0] = (s16)((u16)i2c_smbus_read_word_swapped(drv_client, REG_ACCEL_XOUT_H));
+	g_mpu6050_data.accel_values[1] = (s16)((u16)i2c_smbus_read_word_swapped(drv_client, REG_ACCEL_YOUT_H));
+	g_mpu6050_data.accel_values[2] = (s16)((u16)i2c_smbus_read_word_swapped(drv_client, REG_ACCEL_ZOUT_H));
+	/* gyro */
+	g_mpu6050_data.gyro_values[0] = (s16)((u16)i2c_smbus_read_word_swapped(drv_client, REG_GYRO_XOUT_H));
+	g_mpu6050_data.gyro_values[1] = (s16)((u16)i2c_smbus_read_word_swapped(drv_client, REG_GYRO_YOUT_H));
+	g_mpu6050_data.gyro_values[2] = (s16)((u16)i2c_smbus_read_word_swapped(drv_client, REG_GYRO_ZOUT_H));
+	/* Temperature in degrees C =
+	 * (TEMP_OUT Register Value  as a signed quantity)/340 + 36.53
+	 */
+	//temp = (s16)((u16)i2c_smbus_read_word_swapped(drv_client, REG_TEMP_OUT_H));
+	for (count = 0; count < 1000; count++) {
+		TEMP_OUT_L = i2c_smbus_read_byte_data(drv_client, REG_TEMP_OUT_L);
+		TEMP_OUT_H = i2c_smbus_read_byte_data(drv_client, REG_TEMP_OUT_H);
+
+		temp = (TEMP_OUT_H << 8) | TEMP_OUT_L;
+
+		integer += ((temp + 12420) * 9 / 1700 + 32);
+		rest += ((temp + 12420) * 90 / 17) % 1000;
+
+
+		while (rest >= 1000) {
```

Comment:
Please implement conversation to integer and fractional part without additional loops.